### ISSUE TITLE
Account management

### ISSUE THREAD

**Issue by thequeenofspades** [OurPlanscape/Planscape#3]
Single account based (not Team or Organization based)

**Comment by thequeenofspades** [comment]
This should include basic account management like recovering a forgotten password/username, changing password, deleting account.

**Comment by ayee07** [comment]
Is it related to #483 ?

**Comment by thequeenofspades** [comment]
Yes, but it's larger in scope and should probably be broken down by feature as follows:

- Save additional info from user on registration (first name, last name, department)
- Allow user to edit account information (all fields provided on registration)
- Allow user to add a profile picture
- Allow user to recover an account
- Allow user to delete an account

**Comment by Moon1388** [comment]
Added use cases to [Figma file](https://www.figma.com/file/yJ1vpBMdXuPzs1YoMQR2RR/Planscape-UX---Mar2023?node-id=157%3A70183&t=ePArXitXSWpmb4la-1) in "Account Bug" page. SOME PNGs attached for visual reference. Please refer to Figma for all designs and high level specs. 


![image](https://user-images.githubusercontent.com/125076521/227330621-44ea76c7-41b4-4ef9-b7d8-1f35e48f63ed.png)
![image](https://user-images.githubusercontent.com/125076521/227330718-3bf9d0ae-74ad-47f4-aced-642d062e15e6.png)
![image](https://user-images.githubusercontent.com/125076521/227330855-64f4011e-ceb6-47d7-81c1-ffeb241828f5.png)
![image](https://user-images.githubusercontent.com/125076521/227330925-4871b60b-779c-408a-8af7-133201b42291.png)

**Comment by thequeenofspades** [comment]
Update on this issue:

Account management dialog has been added to the app and allows the user to change their password, update their display name, and delete (disable) their account. See attached PRs for details.

Outstanding items include:

- Account recovery / resetting a password from email
- Verifying user emails
- Account avatars

These won't be finished by March 31st and will remain as work to be completed by the next cohort.

To enable email verification and account recovery, the app needs an email backend configured (see Django's documentation [here](https://docs.djangoproject.com/en/4.1/topics/email/)). I've looked into using a Gmail SMTP server and Gmail Planscape account, but it seems no longer feasible due to Gmail's updated security settings ([related Stackoverflow post](https://stackoverflow.com/questions/72794731/set-django-send-mail-backend-host-with-gmail/72794822#72794822)). It's likely you'll need to use a free (with limited usage) mail sending service like Sendgrid instead.

Account avatars seem feasible but a little tricky to add, because the existing User model in Django doesn't support them and migrating an existing DB to a new user model is prohibitively difficult. You will probably want to create a separate model to store avatars and references to user IDs.
